src/models/geotimeline.py

```diff
@@ -1,6 +1,522 @@
 """Module for analysing multiple GeoGraph objects."""
+from __future__ import annotations
+
+import datetime
+from bisect import bisect_left
+from typing import Callable, Dict, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from src.models.binary_graph_operations import NodeMap, identify_graphs
+from src.models.geograph import GeoGraph
+
+# type alias
+TimeStamp = Union[int, datetime.datetime]
+
+
+class NotCachedError(Exception):
+    """Basic exception for values which were not yet cached."""
+
+
+class TimedGeoGraph(GeoGraph):
```

Comment:
I can see why you would want a time attribute, but what made you choose this way instead of just adding an attribute to `GeoGraph`?

Reply:
True, at the time of creation I wasn't sure if I was going to exted a timed geograph more, I guess I just wanted to have a play with polymorphism 😆  
Not really needed here I agree, can get rid of it in the next update!